include/mbedtls/gcm.h

```diff
@@ -113,21 +113,35 @@ int mbedtls_gcm_setkey( mbedtls_gcm_context *ctx,
  *                  the same as input buffer. If the buffers overlap, the output
  *                  buffer must trail at least 8 Bytes behind the input buffer.
  *
+ * \warning         When this function performs a decryption, it outputs the
+ *                  authentication tag and does not verify that the data is
+ *                  authentic. You should use this function to perform encryption
+ *                  only. For decryption, use mbedtls_gcm_auth_decrypt() instead.
+ *
  * \param ctx       The GCM context to use for encryption or decryption.
  * \param mode      The operation to perform: #MBEDTLS_GCM_ENCRYPT or
- *                  #MBEDTLS_GCM_DECRYPT.
- * \param length    The length of the input data. This must be a multiple of
- *                  16 except in the last call before mbedtls_gcm_finish().
+ *                  #MBEDTLS_GCM_DECRYPT. Note that during decryption, the
+ *                  tag is not verified. You should use this function only
+ *                  to encrypt data, and use mbedtls_gcm_auth_decrypt()
+ *                  to decrypt.
+ * \param length    The length of the input data, which is equal to the length
+ *                  of the output data.
  * \param iv        The initialization vector.
  * \param iv_len    The length of the IV.
  * \param add       The buffer holding the additional data.
  * \param add_len   The length of the additional data.
- * \param input     The buffer holding the input data.
- * \param output    The buffer for holding the output data.
- * \param tag_len   The length of the tag to generate.
+ * \param input     The buffer holding the input data. Its size is \b length.
+ * \param output    The buffer for holding the output data. It must have room
+ *                  for \b length bytes.
+ * \param tag_len   The length of the tag to generate or verify.
```

Comment:
I think it would be better not to add `or verify` as that might seem to contradict the rest of the documentation which correctly states that the function does not verify. (I get that when decrypting the tag is generated for the purpose of verification, but still, the only thing this function does is generate it.)

Reply:
Oops, I didn't intend to write that, the whole point of my patch is that it doesn't verify! Thanks for spotting this.

Because it's a fairly small standalone patch, I've amended the single commit to make this change (and also the formatting issue reported by @Patater). The old version is archived at https://github.com/gilles-peskine-arm/mbedtls/tree/gcm_doc_fix-2.10-1 and likewise for the backports.